src/model/task_manager/TaskManager.cc

```diff
@@ -86,6 +86,38 @@ OperationResult<TMStatus> TaskManager::Delete(TaskId id) {
   return OperationResult<Status>::Error(Status::kNotPresentId);
 }
 
+OperationResult<TMStatus> TaskManager::AddLabel(TaskId id, Label label) {
+  if (auto it = storage_.tasks.find(id); it != storage_.tasks.end()) {
+    auto& task = it->second;
+    if (std::find_if(task.labels().begin(), task.labels().end(),
+                     [&label](auto& stored_lable) {
+                       return label.name() == stored_lable.name();
+                     }) == task.labels().end()) {
+      auto new_label = task.add_labels();
+      *new_label = std::move(label);
+    }
+    return OperationResult<Status>::Ok();
+  }
+  return OperationResult<Status>::Error(Status::kNotPresentId);
+}
+
+OperationResult<TMStatus> TaskManager::DeleteLabel(TaskId id, Label label) {
+  if (auto it = storage_.tasks.find(id); it != storage_.tasks.end()) {
+    auto& task = it->second;
+    if (auto to_erase = std::find_if(task.labels().begin(), task.labels().end(),
```

Comment:
move to_erase definition out of if statement

Reply:
Why? `to_erase` is only used inside of if block